interview-questions.md

```diff
@@ -8,36 +8,67 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: Rails generate resourse - This create both a model and a controller, Rails generate controller, creates a controller, and Rails generate model creates a new model that you can add specific columns to fill with data. 
+
+Researched answer: Rails generate controller is used to create a controller within the model. You use a specific naming convention to create a path
+'parent_module/controller_name' and a example of this would be rails generate controller Rolodex first_name last_name phone_number address. The generate controller uses specific parameters in the form of generate controller ControllerName Thing1 Thing2... Rails generate model creates a model using a naming convention and fields to be filled in with data. rails generate model Rolodex First_name:string. When creating a new model it must be either CamelCased or under_scored, and an optional list of attribute pairs as arguments.
 
-Researched answer:
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET" location: /students
+action: "GET" location: /students 
+This will retrieve all instances of students in the database, Index 
 
 action: "POST" location: /students
+This will create a new instance of Student in the database, Create
 
 action: "GET" location: /students/new
+This will send an HTML form to the user to send their info into the database creating a new instance, New
 
 action: "GET" location: /students/2
+This will send back one instance of the students (2), Show
 
 action: "GET" location: /students/2/edit
+This will update the instance from the HTML form, Edit
 
 action: "PATCH" location: /students/2
+This is a set of instructions on how to modify this instance in the database Overwrite student with the ID of two., Update
 
 action: "DELETE" location: /students/2
+This will delete that instance from the database, Destroy
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
+1) I can give a name to my to do list.
+
+2) I can add a task to my to do list.
+
+3) I can add a description and details to my task.
+
+4) I can mark my task as completed.
+
+5) I can see my completed tasks.
+
+6) I can see how long the task has been pending. 
+
+7) I can add more to do lists based on over all task (Work)(Chores)(School).
+
+8) I can edit the tasks description or details.
+
+9) I can rearrenge the order of my tasks.
+
+10) I can delete a full to do list. 
```

Comment:
This is exactly whats being looked for. To make it more AGILE copy and paste the words, "As a user..." before each statement.